File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/service/receivers/file/ReceiveStoredFileEvent.java
package com.lasthopesoftware.bluewater.client.stored.service.receivers.file;

import com.namehillsoftware.handoff.promises.Promise;

import java.util.Collection;

public interface ReceiveStoredFileEvent {
	Promise<Void> receive(int storedFileId);

	Collection<String> acceptedEvents();
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/sync/receivers/file/ReceiveStoredFileEvent.kt
package com.lasthopesoftware.bluewater.client.stored.sync.receivers.file

import com.lasthopesoftware.bluewater.client.stored.sync.StoredFileMessage
import com.namehillsoftware.handoff.promises.Promise

interface ReceiveStoredFileEvent {
	fun receive(storedFileId: Int): Promise<Unit>
	fun acceptedEvents(): Collection<Class<out StoredFileMessage>>
}
